__tests__/app.test.js

```diff
@@ -449,4 +449,55 @@ describe("App", () => {
         });
     });
   });
+
+  describe("GET /api", () => {
+    test("200: Responds with endpoints object ", () => {
+      return request(app)
+        .get("/api")
+        .expect(200)
+        .then(({ body, body: { endpoints } }) => {
+          expect(body).toHaveProperty("endpoints");
+          expect(Object.keys(endpoints).length).toBeGreaterThanOrEqual(1);
+          for (let endpoint in endpoints) {
+            const endP = endpoints[endpoint];
+            expect(typeof endP.description).toBe("string");
+            if (endP.queries?.length > 1) {
+              endP.queries.forEach((queries) => {
+                for (let query in queries) {
+                  expect(queries[query]).toHaveProperty(
+                    "validQueries",
+                    expect.any(Array)
+                  );
+                  expect(queries[query]).toHaveProperty(
+                    "default",
+                    expect.any(String)
+                  );
+                }
+              });
+            }
+            if (endP.requestBody && Object.keys(endP.requestBody).length > 1) {
+              for (let key in endP.requestBody) {
+                if (endP.requestBody[key]?.requirement) {
+                  expect(endP.requestBody[key]).toHaveProperty(
+                    "requirement",
+                    expect.any(String)
+                  );
+                }
+                expect(endP.requestBody[key]).toHaveProperty(
+                  "dataType",
+                  expect.any(String)
+                );
+                expect(endP.requestBody[key]).toHaveProperty(
+                  "required",
+                  expect.any(Boolean)
+                );
+              }
+            }
+            if (endP.exampleResponse) {
+              expect(typeof endP.exampleResponse).toBe("object");
+            }
+          }
```

Comment:
Because you're not querying a db and instead are reading a file - I think it's perfectly acceptable to check that the object has the property of the endpoints within. And you can also check the length. I think this way is really creative, but could be hard to read for a developer coming in fresh.